### PR TITLE
Added hip-clang support

### PR DESCRIPTION
**Pull request by akilaMD**
This PR adds CI testing for our new hip-clang compiler. For a period of time, we'll have to support both the hcc and the hip-clang compilers, thus we'll have to test both in CI.

### CODE REVIEW

**Comment by ntrost57** [comment]
I have pushed some hip-clang related fixes to develop branch. Can you merge latest develop branch changes into your PR?

**Comment by ntrost57** [comment]
@akilaMD 
* PR #109 and PR #107 should fix the hip-clang build issues
* Commit 4cd3f10 should fix the jenkins permission issue
Can you please update PR #108.

**Comment by akilaMD** [comment]
@ntrost57 

I've made those changes, but having some issues with running CentOS tests, getting a segmentation fault.

File: docker/dockerfile-build-centos [hunk 1]
```diff
@@ -37,8 +35,8 @@ RUN echo '#!/bin/bash' | tee /etc/profile.d/devtoolset7.sh && echo \
 # docker pipeline runs containers with particular uid
 # create a jenkins user with this specific uid so it can use sudo priviledges
 # Grant any member of sudo group password-less sudo privileges
-RUN useradd --create-home -u ${user_uid} -o -G wheel --shell /bin/bash jenkins && \
-    echo '%wheel ALL=(ALL) NOPASSWD:ALL' | tee /etc/sudoers.d/sudo-nopasswd && \
+RUN useradd --create-home -u ${user_uid} -o -G video --shell /bin/bash jenkins && \
+    echo '%video ALL=(ALL) NOPASSWD:ALL' | tee /etc/sudoers.d/sudo-nopasswd && \
```
Comment (saadrahim) [inline docker/dockerfile-build-centos]: why did you need the video group?

File: docker/dockerfile-build-ubuntu-rock [hunk 2]
```diff
@@ -18,8 +18,6 @@ RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-ins
     build-essential \
     ca-certificates \
     git \
-    gcc \
-    g++ \
     make \
     cmake \
     clang-format-3.8 \
```
Comment (saadrahim) [inline docker/dockerfile-build-ubuntu-rock]: do we need to remove clang-format as well?

Comment (ntrost57) [reply docker/dockerfile-build-ubuntu-rock]: guess we can remove it, but shouldnt break anything

File: docker/dockerfile-build-centos (hunk 1 above)
Comment (akilaMD) [reply docker/dockerfile-build-centos]: There was an error complaining about not having access to the GPU. I thought that meant that a non-sudo user couldn't access the GPU, so I switched this to wheel. That then caused permissions issues when running tests. I'm switching it back to the video group because the develop branch is passing CI right now, and the user is being added to the video group there.

File: docker/dockerfile-build-ubuntu-rock (hunk 2 above)
Comment (akilaMD) [reply docker/dockerfile-build-ubuntu-rock]: removing that now
